pkg/controller/operators/olm/operator.go

```diff
@@ -45,340 +38,79 @@ var (
-	op.gcQueueIndexer = csvGCQueueIndexer
-
-	// Set up watch on deployments
-	depHandlers := &cache.ResourceEventHandlerFuncs{
-		// TODO: pass closure that forgets queue item after calling custom deletion handler.
-		DeleteFunc: op.handleDeletion,
-	}
-	for _, namespace := range namespaces {
-		logger.WithField("namespace", namespace).Infof("watching deployments")
-		depInformer := informers.NewSharedInformerFactoryWithOptions(opClient.KubernetesInterface(), wakeupInterval, informers.WithNamespace(namespace)).Apps().V1().Deployments()
-		op.lister.AppsV1().RegisterDeploymentLister(namespace, depInformer.Lister())
-
-		// Register queue and QueueInformer
-		queueName := fmt.Sprintf("%s/csv-deployments", namespace)
-		depQueue := workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), queueName)
-		depQueueInformer := queueinformer.NewInformer(depQueue, depInformer.Informer(), op.syncObject, depHandlers, queueName, metrics.NewMetricsNil(), logger)
-		op.RegisterQueueInformer(depQueueInformer)
-	}
-
-	// Create an informer for the operator group
-	for _, namespace := range namespaces {
-		logger.WithField("namespace", namespace).Infof("watching OperatorGroups")
-		sharedInformerFactory := externalversions.NewSharedInformerFactoryWithOptions(crClient, wakeupInterval, externalversions.WithNamespace(namespace))
-		operatorGroupInformer := sharedInformerFactory.Operators().V1().OperatorGroups()
-		op.lister.OperatorsV1().RegisterOperatorGroupLister(namespace, operatorGroupInformer.Lister())
-
-		// Register queue and QueueInformer
-		queueName := fmt.Sprintf("%s/operatorgroups", namespace)
-		operatorGroupQueue := workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), queueName)
-		operatorGroupQueueInformer := queueinformer.NewInformer(operatorGroupQueue, operatorGroupInformer.Informer(), op.syncOperatorGroups, &cache.ResourceEventHandlerFuncs{DeleteFunc: op.operatorGroupDeleted}, queueName, metrics.NewMetricsNil(), logger)
-		op.RegisterQueueInformer(operatorGroupQueueInformer)
-		op.ogQueueSet.Set(namespace, operatorGroupQueue)
-	}
-
-	return op, nil
-}
-
-func (a *Operator) syncAPIService(obj interface{}) (syncError error) {
+func (o *Operator) syncAPIService(obj interface{}) (syncError error) {
 	apiSvc, ok := obj.(*apiregistrationv1.APIService)
 	if !ok {
-		a.Log.Debugf("wrong type: %#v", obj)
+		o.Log.Debugf("wrong type: %#v", obj)
 		return fmt.Errorf("casting APIService failed")
 	}
 
-	logger := a.Log.WithFields(logrus.Fields{
+	logger := o.Log.WithFields(logrus.Fields{
 		"id":     queueinformer.NewLoopID(),
-		"apiSvc": apiSvc.GetName(),
+		"name": apiSvc.GetName(),
 	})
-	logger.Info("syncing APIService")
+	logger.Info("syncing apiservice")
 
 	if name, ns, ok := ownerutil.GetOwnerByKindLabel(apiSvc, v1alpha1.ClusterServiceVersionKind); ok {
-		_, err := a.lister.CoreV1().NamespaceLister().Get(ns)
+		_, err := o.Lister.CoreV1().NamespaceLister().Get(ns)
```

Comment:
why did Lister and Client become public?

Reply:
I thought it makes using `Operator` as a "base" for other operators easier. Should this be exposed with a public method instead, or is this a sign that the way they're being composed is too rigid?

Reply:
I think it just depends on how you want to re-use `Operator`. I'm not sure that `OperatorClient` and `OperatorLister` are useful outside of olm, for example.